src/main/java/com/gudimov/movieland/web/controller/MovieController.java

```diff
@@ -1,98 +1,99 @@
+        log.info("Sending request to get all movie");
+        Optional<SortOrder> sortOrderRating = Optional.ofNullable(SortOrder.getSortOrderByDirection(ratingSort));
+        Optional<SortOrder> sortOrderPrice = Optional.ofNullable(SortOrder.getSortOrderByDirection(priceSort));
+        validatorSortParams.validate(sortOrderRating, sortOrderPrice);
+        List<Movie> movies = movieService.getAll(sortOrderRating, sortOrderPrice);
+        List<MovieDto> movieDtos = MovieDto.ConvertEntityListToDtoList(movies);
+        String movieJson = JsonJacksonConverter.itemToJson(movieDtos);
+        log.info("Movie {} is received", movieJson);
+
+        return new ResponseEntity<>(movieJson, HttpStatus.OK);
+
+    }
+
+    @RequestMapping(value = "/random")
+    @ResponseBody
+    public ResponseEntity<String> getRandomMovie() {
+        log.info("Sending request to get random movie");
+        List<Movie> movies = movieService.getRandom();
+        List<MovieRandomDto> movieDtos = new ArrayList<>();
+        for (Movie movie : movies) {
+            movieDtos.add(new MovieRandomDto(movie));
+        }
+        String movieJson = JsonJacksonConverter.itemToJson(movieDtos);
+        log.info("Movie {} is received", movieJson);
+        return new ResponseEntity<>(movieJson, HttpStatus.OK);
+
+    }
+
+    @RequestMapping(value = "/genre/{genreId}")
+    @ResponseBody
+    public ResponseEntity<String> getByGenreId(@PathVariable int genreId,
+                                               @RequestParam(value = "rating", required = false) String ratingSort,
+                                               @RequestParam(value = "price", required = false) String priceSort) {
+        log.info("Sending request to get movie by genre id = {}", genreId);
+        Optional<SortOrder> sortOrderRating = Optional.ofNullable(SortOrder.getSortOrderByDirection(ratingSort));
+        Optional<SortOrder> sortOrderPrice = Optional.ofNullable(SortOrder.getSortOrderByDirection(priceSort));
+        validatorSortParams.validate(sortOrderRating, sortOrderPrice);
+        List<Movie> movies = movieService.getByGenreId(genreId, sortOrderRating, sortOrderPrice);
+        List<MovieDto> movieDtos = MovieDto.ConvertEntityListToDtoList(movies);
+        String movieJson = JsonJacksonConverter.itemToJson(movieDtos);
+        log.info("Movie {} is received", movieJson);
+
+        return new ResponseEntity<>(movieJson, HttpStatus.OK);
+    }
+
+    @RequestMapping(value = "/{movieId}")
+    @ResponseBody
+    public ResponseEntity<String> getById(@PathVariable int movieId,
+                                          @RequestParam(value = "currency", required = false) String currency) {
+        log.info("Sending request to get movie by id = {}", movieId);
+        Optional<CurrencyCode> currencyCode = Optional.ofNullable(CurrencyCode.getCurrencyByCode(currency));
+        Movie movie = movieService.getById(movieId, currencyCode);
```

Comment:
you'll have a problem with current architecture here when implement cache for movie: different CCYs could appear there. So on service layer all movies should be with default CCY, after that put it into cache, and then check CCY parameter on controller level (enrich if exists).